digital_image_processing/index_calculation.py

```diff
@@ -1,651 +1,525 @@
+            :return: index
+        """
+        return red / (nir + red + green)
+
+    def NGRDI(self):
+        """
+            Normalized Difference Green/Red Normalized green red difference index, Visible Atmospherically Resistant Indices Green (VIgreen)
+            https://www.indexdatabase.de/db/i-single.php?id=390
+            :return: index
+        """
+        return (green - red) / (green + red)
+
+    def RI(self):
+        """
+            Normalized Difference Red/Green Redness Index
+            https://www.indexdatabase.de/db/i-single.php?id=74
+            :return: index
+        """
+        return (red - green) / (red + green)
+
+    def S(self):
+        """
+            Saturation
+            https://www.indexdatabase.de/db/i-single.php?id=77
+            :return: index
+        """
+        max = np.max([np.max(red), np.max(green), np.max(blue)])
+        min = np.min([np.min(red), np.min(green), np.min(blue)])
+        return (max - min) / max
+
+    def IF(self):
+        """
+            Shape Index
+            https://www.indexdatabase.de/db/i-single.php?id=79
+            :return: index
+        """
+        return (2 * red - green - blue) / (green - blue)
+
+    def DVI(self):
+        """
+            Simple Ratio NIR/RED Difference Vegetation Index, Vegetation Index Number (VIN)
+            https://www.indexdatabase.de/db/i-single.php?id=12
+            :return: index
+        """
+        return nir / red
+
+    def TVI(self):
+        """
+            Transformed Vegetation Index
+            https://www.indexdatabase.de/db/i-single.php?id=98
+            :return: index
+        """
+        return (self.NDVI() + 0.5) ** (1 / 2)
+
+    def NDRE(self):
+        return (nir - redEdge) / (nir + redEdge)
+
+
+"""
+# genering a random matrices to test this class
+red     = np.ones((1000,1000, 1),dtype="float64") * 46787
+green   = np.ones((1000,1000, 1),dtype="float64") * 23487
+blue    = np.ones((1000,1000, 1),dtype="float64") * 14578
+redEdge = np.ones((1000,1000, 1),dtype="float64") * 51045
+nir     = np.ones((1000,1000, 1),dtype="float64") * 52200
+
+# Examples of how to use the class
+
+# instantiating the class 
+cl = IndexCalculation()
+
+# instantiating the class with the values
+#cl = indexCalculation(red=red, green=green, blue=blue, redEdge=redEdge, nir=nir)
+
+# how set the values after instantiate the class cl, (for update the data or when dont instantiating the class with the values)
+cl.setMatrices(red=red, green=green, blue=blue, redEdge=redEdge, nir=nir)
+
+# calculating the indices for the instantiated values in the class
+    # Note: the CCCI index can be changed to any index implemented in the class.
+indexValue_form1    = cl.calculation("CCCI").astype(np.float64)
+indexValue_form2    = cl.CCCI()
+
+# calculating the index with the values directly -- you can set just the values preferred -- note: the *calculation* fuction performs the function *setMatrices*
```

Comment:
These lines should be wrapped so that they can be read on GitHub with left-right scrolling.